packages/gatsby/src/services/initialize.ts

```diff
@@ -162,6 +162,21 @@ export async function initialize({
 
   activity.end()
 
+  if (
+    process.env.gatsby_executing_command === `develop` &&
+    process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND
+  ) {
+    if (isCI()) {
+      process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND = undefined
+      reporter.warn(
+        `Experimental Query on Demand feature is not available in CI environment. Continuing with regular mode.`
+      )
+    } else {
+      reporter.info(`Using experimental Query on Demand feature`)
+      telemetry.trackFeatureIsUsed(`QueryOnDemand`)
+    }
+  }
```

Comment:
Doing this here (after reading `gatsby-config` (s)) allow for adding

```
process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND = true
```

in the `gatsby-config.js` file which makes it easier to work with env var based feature flag

Reply:
I think I actually should change this a bit if/else conditions a bit to make sure I set `process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND = undefined` in builds just in case we didn't put all the checks that would only do this in `gatsby develop`?